### ISSUE TITLE
arm64 linux runtime SIGSEGVs on console helloworld

### ISSUE THREAD

**Issue by t-mustafin** [dotnet/runtime#77217]
### Description

All arm64 programs fails with SIGSEGV on release build of runtime.

Backtrace:
```
(lldb) run hello/hello.dll
There is a running process, kill it and restart?: [Y/n] y
Process 514688 exited with status = 9 (0x00000009) 
Process 515318 launched: '/root/netcoreapp/corerun' (aarch64)
Process 515318 stopped
* thread #1, name = 'corerun', stop reason = signal SIGSEGV: address access protected (fault address: 0x7fb770a288)
    frame #0: 0x0000007fb76ea288
->  0x7fb76ea288: ldr    x11, #0x20000
    0x7fb76ea28c: br     x11
    0x7fb76ea290: ldr    x12, #0x20000
    0x7fb76ea294: ldr    x11, #0x20080
(lldb) bt
* thread #1, name = 'corerun', stop reason = signal SIGSEGV: address access protected (fault address: 0x7fb770a288)
  * frame #0: 0x0000007fb76ea288
    frame #1: 0x0000007ff7629450 libcoreclr.so`MethodDescCallSite::CallTargetWorker(unsigned long long const*, unsigned long long*, int) at callhelpers.cpp:67:5
    frame #2: 0x0000007ff76293f4 libcoreclr.so`MethodDescCallSite::CallTargetWorker(this=0x0000007fffffe2c0, pArguments=0x0000007fffffe3c8, pReturnValue=0x0000000000000000, cbReturnValue=0) at callhelpers.cpp:562
    frame #3: 0x0000007ff754e694 libcoreclr.so`CorHost2::CreateAppDomainWithManager(char16_t const*, unsigned int, char16_t const*, char16_t const*, int, char16_t const**, char16_t const**, unsigned int*) [inlined] MethodDescCallSite::Call(this=<unavailable>, pArguments=<unavailable>) at callhelpers.h:458:9
    frame #4: 0x0000007ff754e680 libcoreclr.so`CorHost2::CreateAppDomainWithManager(this=0x00000055555919c0, wszFriendlyName=<unavailable>, dwFlags=<unavailable>, wszAppDomainManagerAssemblyName=<unavailable>, wszAppDomainManagerTypeName=<unavailable>, nProperties=3, pPropertyNames=0x0000005555592e80, pPropertyValues=0x0000005555592ec0, pAppDomainID=0x0000005555575060) at corhost.cpp:603
    frame #5: 0x0000007ff7510590 libcoreclr.so`::coreclr_initialize(exePath=0x0000005555592f90, appDomainFriendlyName=<unavailable>, propertyCount=<unavailable>, propertyKeys=<unavailable>, propertyValues=<unavailable>, hostHandle=<unavailable>, domainId=<unavailable>) at exports.cpp:254:16
    frame #6: 0x000000555555599c corerun`main [inlined] run(config=0x0000007fffffe5f8) at corerun.cpp:348:14
    frame #7: 0x00000055555544c0 corerun`main(argc=<unavailable>, argv=<unavailable>) at corerun.cpp:563
    frame #8: 0x0000007ff7b7aff4 libc.so.6`__libc_start_main(main=(corerun`main at corerun.cpp:555), argc=2, argv=0x0000007ffffff3a8, init=<unavailable>, fini=<unavailable>, rtld_fini=<unavailable>, stack_end=<unavailable>) at libc-start.c:308:16
    frame #9: 0x00000055555538b4 corerun`_start at start.S:92
(lldb)
``` 

### Reproduction Steps

```
./corerun ./hello.dll
```

### Expected behavior

```
./corerun ./hello.dll
Hello
```

### Actual behavior

```
[rpi4 ~]# ./netcoreapp/corerun ./hello/hello.dll 
Segmentation fault (core dumped)
[rpi4 ~]#
```

### Known Workarounds

No problem occurs on debug build.

### Configuration

Runtime main branch, commit: b6852e01fd90d409b5a4267be770668a5d212298
OS: Tizen7/Unified 7.0.0 (aarch64)
Arch: arm64
Configuration: Release

**Comment by t-mustafin** [comment]
cc @alpencolt @gbalykov

**Comment by janvorli** [comment]
Yes, this is a dup of #75374. @t-mustafin you'll need to install LLD linker to fix the issue or cherry pick my fix #77275 into the .NET 7 sources.

**Comment by t-mustafin** [comment]
@janvorli , thank you.
